presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestingPrestoClient.java

```diff
@@ -104,7 +105,8 @@ public void close()
             }
 
             if (error.getFailureInfo() != null) {
-                throw error.getFailureInfo().toException();
+                RuntimeException remoteException = error.getFailureInfo().toException();
+                throw new RuntimeException(Optional.ofNullable(remoteException.getMessage()).orElseGet(remoteException::toString), remoteException);
```

Comment:
I think this should be
```java
throw new RuntimeException(remoteException);
```
This will use the custom `toString()` of `FailureException` which prints the original exception's class name. Is there a reason we don't want that?

Reply:
It could be. However that means exception _message_ would now contain "com....SemanticException: something-something" instead of just "something-something". Tests using `@Test(expectedExceptionMsg)` would no longer pass.

Reply:
Makes sense